lib/smack/SmackRep.cpp

```diff
@@ -98,6 +171,16 @@ string SmackRep::memReg(unsigned idx) {
   return s.str();
 }
 
+string SmackRep::memReg(unsigned idx, bool safety, unsigned type) {
```

Comment:
I would say that memReg should  concatenate with int_type() instead of memType(). Is this what you mean? 

The reason is to separate bytes and i8 is to simplify program logic by dividing memory regions to safe (i8, i16, i32..) and unsafe (bytes) ones. I'll try to mix M.X.bytes and M.X.i8 into single entity.